x/metadata/keeper/scope.go

```diff
@@ -88,22 +67,81 @@ func (k Keeper) IterateScopesForScopeSpec(ctx sdk.Context, scopeSpecID types.Met
 	return nil
 }
 
-// GetScope returns the scope with the given id.
-func (k Keeper) GetScope(ctx sdk.Context, id types.MetadataAddress) (scope types.Scope, found bool) {
+// GetScope returns the scope with the given id. The ValueOwnerAddress field will always be empty from this method.
+// See also: GetScopeWithValueOwner, PopulateScopeValueOwner and GetScopeValueOwner.
+func (k Keeper) GetScope(ctx sdk.Context, id types.MetadataAddress) (types.Scope, bool) {
 	if !id.IsScopeAddress() {
-		return scope, false
+		return types.Scope{}, false
 	}
 	store := ctx.KVStore(k.storeKey)
 	b := store.Get(id.Bytes())
 	if b == nil {
 		return types.Scope{}, false
 	}
-	k.cdc.MustUnmarshal(b, &scope)
-	return scope, true
+	return k.mustReadScopeBz(b), true
+}
+
+// readScopeBz will unmarshal the given bytes into a Scope.
+// The ValueOwnerAddress will always be empty. See also: GetScopeValueOwner, PopulateScopeValueOwner.
+func (k Keeper) readScopeBz(bz []byte) (types.Scope, error) {
+	var scope types.Scope
+	err := k.cdc.Unmarshal(bz, &scope)
+	// In the viridian upgrade, we switched to using the bank module to track the ValueOwnerAddress,
+	// but we didn't update all the scopes to remove the data from that field. Any value in that field
+	// in state is therefore not to be trusted (probably out of date), so we clear it out here.
+	scope.ValueOwnerAddress = ""
+	return scope, err
+}
+
+// mustReadScopeBz will unmarshal the given bytes into a Scope, panicking if there's an error.
+// The ValueOwnerAddress will always be empty. See also: GetScopeValueOwner, PopulateScopeValueOwner.
+func (k Keeper) mustReadScopeBz(bz []byte) types.Scope {
+	scope, err := k.readScopeBz(bz)
+	if err != nil {
+		panic(err)
+	}
+	return scope
+}
+
+// GetScopeWithValueOwner will get a scope from state and also look up and set its value owner field.
+func (k Keeper) GetScopeWithValueOwner(ctx sdk.Context, id types.MetadataAddress) (scope types.Scope, found bool) {
+	scope, found = k.GetScope(ctx, id)
+	if found {
+		k.PopulateScopeValueOwner(ctx, &scope)
+	}
+	return scope, found
+}
+
+// PopulateScopeValueOwner will look up and set the ValueOwnerAddress in the provided scope.
+func (k Keeper) PopulateScopeValueOwner(ctx sdk.Context, scope *types.Scope) {
+	vo, err := k.GetScopeValueOwner(ctx, scope.ScopeId)
+	if err == nil && len(vo) > 0 {
+		scope.ValueOwnerAddress = vo.String()
+	} else {
+		scope.ValueOwnerAddress = ""
+	}
```

Comment:
In all cases, an error from `GetScopeValueOwner` means that there is no value owner for the scope. `PopulateScopeValueOwner` was designed as a quick and easy way to look up and set the `ValueOwnerAddress` in a scope right after reading the scope from state. In all places where `PopulateScopeValueOwner` is called, the error would just be ignored anyway. So it's better to handle the error from inside `PopulateScopeValueOwner` like it is currently being done.

It is highly unlikely that `GetScopeValueOwner` returns an error anyway, but in the places where we do care about such an error, `GetScopeValueOwner` is called directly (instead of using `PopulateScopeValueOwner`).

Further, since `PopulateScopeValueOwner` is altering a field in a provided argument; having it return an error means you don't really know what happened to the content of the `scope`. For example, with your suggested modification, the `ValueOwnerAddress` would be unchanged if there were an error. However, it'd be better if it cleared it out in such a case. That's an easy tweak to make, but it highlights the ambiguity that is caused by having `PopulateScopeValueOwner` return an error. Without it returning an error, you can be certain that the `scope` is in the correct state after giving it to `PopulateScopeValueOwner`. With it returning an error, not only does the caller have to handle the error (which is always to just ignore it), but they might also have to update the `scope`.

It's just not good for this to return an error.